lib/destroyed_at/has_one_association.rb

```diff
@@ -2,7 +2,7 @@ module DestroyedAt
   module HasOneAssociation
     def delete(method = options[:dependent])
       if load_target && method == :destroy
-        DestroyedAt.destroy_target_of_association(target, owner)
+        DestroyedAt.destroy_target_of_association(owner, target)
```

Comment:
Thanks for catching this @anarchocurious. It looks like this was also a problem in our `BelongsToAssociations`.